### PR TITLE
feat(pops): `L2OutputOracle` updates

### PR DESCRIPTION
**Pull request by clabby**
# Overview

Adds the required `L2OutputOracle` changes for the permissionless output proposals project.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* feat/pops-contracts
  * **PR #5723** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/5723" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/5723?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol [hunk 1]
```diff
@@ -131,45 +140,49 @@ contract L2OutputOracle is Initializable, Semver {
...
+        delete l2Outputs[index - 1];
 
-        // Use assembly to delete the array elements because Solidity doesn't allow it.
-        assembly {
-            sstore(l2Outputs.slot, _l2OutputIndex)
-        }
-
-        emit OutputsDeleted(prevNextL2OutputIndex, _l2OutputIndex);
+        uint256 l2OutputsLength = l2Outputs.length;
+        emit OutputsDeleted(l2OutputsLength, l2OutputsLength);
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:181]: We should be able to get away with breaking the ABI here to be honest to create an event that makes more sense.

```solidity
bytes32 output = l2Outputs[index - 1];
delete l2Outputs[index - 1];
emit OutputDeleted(output, _l2BlockNumber, msg.sender);
```

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol [hunk 2]
```diff
@@ -183,14 +196,12 @@ contract L2OutputOracle is Initializable, Semver {
         uint256 _l1BlockNumber
     ) external payable {
         require(
-            msg.sender == PROPOSER,
-            "L2OutputOracle: only the proposer address can propose new outputs"
+            msg.value >= OUTPUT_BOND_COST,
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:199]: Didn't we decide on using the simple bond scheme as seen in the [specs](https://github.com/ethereum-optimism/optimism/blob/develop/specs/bond-manager.md#simple-bond)?

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol [hunk 3]
```diff
@@ -217,7 +228,12 @@ contract L2OutputOracle is Initializable, Semver {
             );
         }
 
-        emit OutputProposed(_outputRoot, nextOutputIndex(), _l2BlockNumber, block.timestamp);
+        // Post the bond to the bond manager
+        BOND_MANAGER.post{ value: msg.value }({
+           _bondId: bytes32(abi.encode(_l2BlockNumber)),
+           _bondOwner: msg.sender,
+           _minClaimHold: FINALIZATION_PERIOD_SECONDS
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:235]: An invariant exists that there must always be a bond while the output is mutable (able to be removed)

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol [hunk 4]
```diff
@@ -334,7 +369,7 @@ contract L2OutputOracle is Initializable, Semver {
      * @return Next L2 block number.
      */
     function nextBlockNumber() public view returns (uint256) {
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:371]: I lean towards deprecating this function as its semantics change, its obvious if you read the `@notice` and then look at the new implementation

Comment (tynes) [reply packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:371]: Same for `latestBlockNumber`

File: packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol (hunk 2 above)
Comment (refcell) [reply packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol:199]: Correct, these are very old changes now

**Pull request closed without merging**